### PR TITLE
Adding an English translation version: chp. 0 & 1 & references

### PR DESCRIPTION
**Pull request by YuelinXin**
Added an English translation version of this book, namely chapter 0, 1 & references. Manually translated, with the help of GitHub Copilot & ChatGPT. Spelling checked with VSCode Spell Bot.

**All of the contents** are directly translated from the original Chinese version, with very small tweaks where English and Chinese don't match. I tried my best to convey the author's ideas and explanations as clearly as possible, please bear with me if there are any mistake.

In exclusion of several parts that contains the author's personal experience, namely the "Acknowledgment" and "About the Author" part of chapter 0, all other parts are fully translated.

All the illustrations & animations are not translated.

### CODE REVIEW

File: docs/en/chapter_preface/suggestions.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+## Learning With Text + Illustrations
+
+Videos and images convey information in a denser and more structured way than text, making them easier to understand. In this book, the key and difficult parts will be presented mainly with animations and illustrations, and the purpose of the text is to explain and supplement the animations and illustrations.
+
+When reading this book, if you find that a section contains an animation or illustration, **it is recommended that you use the illustrations as the main reference**, and try to map the text content (usually above the image) to the content in the illustrations to get a better understanding.
+
+![algorithm_animation](suggestions.assets/algorithm_animation.gif)
+
+## Learning by Doing
```
Comment (krahets) [inline docs/en/chapter_preface/suggestions.md:15]: How about `Learning by Practicing`

**Review by krahets — COMMENTED** [review]
LGTM! Thanks for the great work! Could you remove all the figures? I'll translate and add them.

**Comment by YuelinXin** [comment]
Done :)

**Review by krahets — APPROVED** [review]
Thanks! Now merge it. I created a new branch `en-dev`. Let's push new changes to it. 🥂

**Comment by krahets** [comment]
@YuelinXin Temporarily removed the `en-dev` branch because the Chinese version has too many updates. Please create a pull request containing the refreshed version. Thanks!

**Comment by YuelinXin** [comment]
Will merge the newest changes in the Chinese version then try to add PR again.

**Pull request merged**